### PR TITLE
server: precreate some labal metrics

### PR DESCRIPTION
**Pull request by july2993**
for some frequently use metrics, we can pre create the Metric for later use, this avoid with_label_values() every time, but maybe  needless here...it help little

### CODE REVIEW

**Comment by siddontang** [comment]
Thank you very much. @july2993 

This reduces the hash and lock operations, it is useful. :-)

**Comment by siddontang** [comment]
PTAL @disksing @BusyJay

File: src/server/service/kv.rs [hunk 1]
```diff
@@ -57,6 +58,57 @@ pub struct Service<T: RaftStoreRouter + 'static> {
...
+    kv_delete_range: Histogram,
+    raw_get: Histogram,
+    raw_scan: Histogram,
+    raw_put: Histogram,
+    raw_delete: Histogram,
+    coprocessor: Histogram,
+    mvcc_get_by_key: Histogram,
+    mvcc_get_by_start_ts: Histogram,
+    split_region: Histogram,
+}
```
Comment (disksing) [inline src/server/service/kv.rs]: Why not use `HashMap<&str, Histogram>`?

Comment (siddontang) [reply src/server/service/kv.rs]: maybe using fields is faster and more certain than HashMap.

**Comment by breeswish** [comment]
Can we use LocalXxxVec here?

**Comment by siddontang** [comment]
> Can we use LocalXxxVec here

No, now we have no way to flush the local metrics here.

File: src/server/service/kv.rs (hunk 1 above)
Comment (july2993) [reply src/server/service/kv.rs]: and here if using HashMap, we still need to init it one by one, and no need to create some  metrics dynamic, so i use fields

**Comment by disksing** [comment]
LGTM.

**Comment by breeswish** [comment]
Amazing! I just benchmarked your PR and observed a ~7% performance improvement for raw kv commands :tada: Please resolve conflicts

**Comment by siddontang** [comment]
Thank @july2993 

There are still some MetricVectors in our codes, it is very appreciated that you can help us improve them.

**Comment by july2993** [comment]
@breeswish , thanks for the benchmark. well...this much better than my expect, wonder about the benchmark detail. maybe i will take a more look later.

**Comment by july2993** [comment]
@siddontang, at a first look, the GRPC_MSG_FAIL_COUNTER.with_label_values in kv.rs can be avoid easyly too,  but i think it will seldomly reach the code in a stable cluster(maybe except the kv_commit api),  so i did not change it first when i do this pr.
another is the KV_COMMAND_COUNTER_VEC.with_label_values in storage/mod.rs, one rpc one call.

**Comment by breeswish** [comment]
@disksing there are "endpoint read failed" in jenkins unit test. Could you take a look?

**Comment by disksing** [comment]
@breeswish I'm sorry I have no clue... @overvenus @BusyJay would you want to lend a hand?

**Comment by BusyJay** [comment]
@hicqu PTAL

**Comment by breeswish** [comment]
LGTM

**Pull request merged**